api/helpers.go

```diff
@@ -66,3 +67,40 @@ func (a *API) requestAud(ctx context.Context, r *http.Request) string {
 	// Finally, return the default of none of the above methods are successful
 	return config.JWT.Aud
 }
+
+func (a *API) isEmailBlacklisted(email string) bool {
+	globalConfig := a.config
+	if a.blacklist.UpdateNeeded() {
+		a.blacklist.UpdateFromURL(globalConfig.EmailBlacklist.URL)
+	}
+	return a.blacklist.EmailBlacklisted(email)
+}
+
+func (a *API) isPasswordValid(password string) bool {
+	re_min_nums := regexp.MustCompile("[0-9]")
```

Comment:
3 things 
- regex's should be compiled once whenever possible, especially in ones that don't have dynamic parts.
- golang uses camelCase not snake_case
- instead of using `[0-9]` I'd tend to use `[[:digit:]]` and the `[[:upper:]]`

it also looks like you're not requiring lowercase characters.

Reply:
@rybit Oy, sometimes my Ruby background sneaks in.

So something like this would be preferrable?

```go
var minNumsRegexp = regexp.MustCompile("[[:digit:]]")

func (a *API) isPasswordValid(password string) bool {
  // ...
}
```

Also I noticed in `gotrue/api/api.go` that there was a regex compiled with a suffix of `Regexp` so I changed my variable name to match to be consistent.

Not sure if there's another way to accomplish this.